client/cli/src/main.rs

```diff
@@ -228,27 +237,46 @@ impl Error for CliError {}
 
 #[derive(Parser)]
 #[command(version)]
-struct Cli {
+struct Opts {
     /// The working directory of the client containing the config and db
     #[arg(long = "workdir")]
-    workdir: PathBuf,
+    workdir: Option<PathBuf>,
+
     #[clap(subcommand)]
     command: Command,
 }
 
-#[derive(Parser)]
-#[command(version)]
-struct CliNoWorkdir {
-    #[clap(subcommand)]
-    command: CommandNoWorkdir,
-}
+impl Opts {
+    fn workdir(&self) -> &Path {
+        self.workdir
+            .as_ref()
+            .or_terminate(CliErrorKind::IOError, "`--workdir=` argument not set.")
```

Comment:
nit: I think there are ways in clap to model dependencies between arguments/argument groups that would allow using clap's built-in help system. Not for you but a good first issue I'll create after merging.

Reply:
I *think* if we split the commands that need a work dir and the ones who don't into two enums we could do it in a non-brittle way, but I agree this might be overdoing it.

Reply:
We were doing just that, and someone put `decode-connect-info` in the wrong one. I don't blame them - my point is exactly that it's additional complication that is easy to forget about, just to display the exactly same error, just by a different part of code (clap vs our code).